ecommerce/views/v0/__init__.py

```diff
@@ -0,0 +1,79 @@
+"""
+MITxOnline ecommerce views
+"""
+
+import pytz
+import datetime
+
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from rest_framework.viewsets import ReadOnlyModelViewSet
+from rest_framework.pagination import LimitOffsetPagination
+from django.db.models import Q, Count
+
+from courses.models import (
+    CourseRun,
+    Course,
+    Program,
+)
+
+from ecommerce.serializers import ProductSerializer
+from ecommerce.models import Product
+
+class ProductsPagination(LimitOffsetPagination):
+    default_limit = 2
+    limit_query_param = 'l'
+    offset_query_param = 'o'
+    max_limit = 50
+
+class ProductViewSet(ReadOnlyModelViewSet):
+    serializer_class = ProductSerializer
+    pagination_class = ProductsPagination
+
+    def get_queryset(self):
+        now = datetime.datetime.now(tz=pytz.UTC)
+
+        expired_courseruns = CourseRun.objects.filter(
+            enrollment_end__lt=now
+        ).values_list("id", flat=True)
+
+        expired_courses = (
+            Course.objects.annotate(
+                runs=Count("courseruns", filter=~Q(courseruns__in=expired_courseruns))
+            )
+            .filter(runs=0)
+            .values_list("id", flat=True)
+        )
+
+        expired_programs = (
```

Comment:
I am not sure about the terminology? I feel like expired could mean many different things, like it is not enrollable anymore, or it has finished, or things like that. In MM we used the word 'past' to refer to end date being in the past.

Reply:
`unenrollable_program_ids` is probably what I'd go with